src/Playwright/Transport/Connection.cs

```diff
@@ -247,30 +241,31 @@ internal void Dispatch(PlaywrightServerMessage message)
 
                 if (message.Method == "__adopt__")
                 {
-                    Objects.TryGetValue(message.Guid, out var child);
+                    var childGuid = message.Params.Value.GetProperty("guid").GetString();
+                    Objects.TryGetValue(childGuid, out var child);
                     if (child == null)
                     {
-                        throw new PlaywrightException($"Unknown new child: '{message.Guid}'");
+                        throw new PlaywrightException($"Unknown new child: '{childGuid}'");
                     }
                     @object.Adopt((ChannelOwnerBase)child);
+                    return;
                 }
 
                 if (message.Method == "__dispose__")
                 {
-                    @object?.DisposeOwner();
+                    @object.DisposeOwner();
                     return;
                 }
 
-                Objects.TryGetValue(message.Guid, out var obj);
-                obj?.Channel?.OnMessage(message.Method, message.Params);
+                @object.Channel?.OnMessage(message.Method, message.Params);
             }
             catch (Exception ex)
             {
                 DoClose(ex);
             }
         }
 
-        private void CreateRemoteObject(string parentGuid, ChannelOwnerType type, string guid, JsonElement? initializer)
+        private IChannelOwner CreateRemoteObject(string parentGuid, ChannelOwnerType type, string guid, JsonElement? initializer)
```

Comment:
```suggestion
        private IChannelOwner CreateRemoteObject(string parentGuid, ChannelOwnerType type, string guid, JsonElement? initializer)
```
```suggestion
        private void CreateRemoteObject(string parentGuid, ChannelOwnerType type, string guid, JsonElement? initializer)
```
Do I see it correctly that the return value is unused?

Reply:
Yes, the return value is unused. But that is also the case in javascript and java 😄 
I can change it if you want, maybe this should be done everywhere?

https://github.com/microsoft/playwright/blob/d56b27962207de38f3b4427bcf4f2a3d9f8b1077/packages/playwright-core/src/client/connection.ts#L188-L294

https://github.com/microsoft/playwright-java/blob/4d278c391e3c50738ddea6c3e324a4bbbf719d86/playwright/src/main/java/com/microsoft/playwright/impl/Connection.java#L225-L326